R/utils.R

```diff
@@ -209,12 +209,27 @@ activeEncoding <- function(project = getwd()) {
   sub("^Encoding:\\s*", "", encodingLine)
 }
 
-md5sum <- function(path) {
-  # open the file for reading in binary mode (to ensure we treat newlines
-  # literally when computing the md5)
-  con <- base::file(path, open = "rb")
-  on.exit(close(con), add = TRUE)
-
-  # compute md5 sum of contents and return as ordinary characters
-  unclass(as.character(openssl::md5(con)))
+# Returns the MD5 for path as a raw sequence of 16 hexadecimal pairs.
+fileMD5 <- function(path) {
+  if (is.null(path)) {
+    # Use raw(0) rather than the empty string so openssl::md5 returns the has as hex values and not
+    # a concatenated string of hex characters.
+    return(openssl::md5(raw(0)))
```

Comment:
Does this call still work even in FIPS mode? (Why not just use `digest()` here?)

Reply:
Oy. That's my mistake. I intended to make that use `digest` as well. PR shortly.

Reply:
addressed in https://github.com/rstudio/rsconnect/pull/383.